src/common/timezone-picker.js

```diff
@@ -9,90 +9,84 @@ import propTypes from './prop-types'
 import { getXoServerTimezone } from './xo'
 import { Select } from './form'
 
-const XO_SERVER_TIMEZONE = 'xo-server'
+const SERVER_TIMEZONE = 'server'
 
 @propTypes({
   defaultValue: propTypes.string,
   onChange: propTypes.func.isRequired,
+  required: propTypes.bool,
   value: propTypes.string
 })
 export default class TimezonePicker extends Component {
   constructor (props) {
     super(props)
-    this.state.options = map(moment.tz.names(), value => ({ label: value, value }))
-  }
 
-  get value () {
-    const value = this.refs.select.value
-    return (value === XO_SERVER_TIMEZONE) ? null : value
+    this._localTimezone = moment.tz.guess()
+    this._serverTimezone = SERVER_TIMEZONE
   }
 
-  set value (value) {
-    this.refs.select.value = value || XO_SERVER_TIMEZONE
+  componentDidMount () {
+    getXoServerTimezone.then(serverTimezone => {
+      this.setState({
+        timezone: this.props.value || this.props.defaultValue || this._serverTimezone,
+        options: [
+          ...map(moment.tz.names(), value => ({ label: value, value })),
+          {
+            label: _('serverTimezoneOption', {
+              value: serverTimezone
+            }),
+            value: this._serverTimezone
+          }
+        ]
+      })
+    })
   }
 
-  _updateTimezone (value) {
-    this.props.onChange(value)
-  }
-  _handleChange = option => {
-    return this._updateTimezone(
-      !option || option.value === XO_SERVER_TIMEZONE
-        ? null
-        : option.value
-    )
+  componentWillReceiveProps (props) {
+    if (props.value !== this.props.value) {
+      this.setState({ timezone: props.value || this._serverTimezone })
+    }
   }
-  _useServerTime = () => {
-    this._updateTimezone(null)
+
+  get value () {
+    return this.state.timezone === this._serverTimezone ? null : this.state.timezone
   }
-  _useLocalTime = () => {
-    this._updateTimezone(moment.tz.guess())
+
+  set value (value) {
+    this.setState({ timezone: value })
   }
 
-  componentWillMount () {
-    // Use local timezone (Web browser) if no default value.
-    if (this.props.value === undefined) {
-      this._useLocalTime()
+  _onChange = option => {
+    if (option && option.value === this.state.timezone) {
+      return
     }
 
-    getXoServerTimezone.then(serverTimezone => {
-      this.setState({
-        options: [{
-          label: _('serverTimezoneOption', {
-            value: serverTimezone
-          }),
-          value: XO_SERVER_TIMEZONE
-        }].concat(this.state.options),
-        serverTimezone
-      })
-    })
+    this.setState({
+      timezone: option && option.value
+    }, () =>
+      this.props.onChange(option && (option.value === this._serverTimezone ? null : option.value))
```

Comment:
Why waiting for the state to be changed?

And if you do, why not using directly the `state` as the source of truth?